File: sprint1apache/www/site3/calculadora.php
<html>
        <body>
                <h1>Calculadora</h1>
                <p>
                <?php
                        if (isset($_POST["calcular"])) {
                                $num1 = $_POST['num1'];
				$num2 = $_POST['num2'];
				$op = $_POST['op'];
				if (empty($num1) || empty($num2) || !is_numeric($num1) || !is_numeric($num2)) {
					echo "Introduce un número válido por favor";
				} else {
					switch($op) {
						case "+":
							$resultado = $num1 + $num2;
							break;
						case "-":
							$resultado = $num1 + $num2;
							break;
						case "*":
							$resultado = $num1 * $num2;
							break;
						case "/":
							$resultado = $num1 / $num2;
							break;
						default:
							echo "Operación imposible";
					}
				}
				echo "El resultado final de ".$num1." ".$op." ".$num2." es ".$resultado;
                        }
                ?>
                </p>
                <p>Realiza una nueva operación:</p>

                <form action="calculadora.php" method="post">
			<label for="num1">Introduce el primer número: </label>
			<input type="number" id="num1" name="num1" required><br>
			<label for="num2">Introduce el segundo número: </label>
			<input type="number" id="num2" name="num2" required><br>
			<label for="op">Escoge un operación: </label>
			
			<select id="op" name="op">
				<option value="+">+</option>
				<option value="-">-</option>
				<option value="*">*</option>
				<option value="/">/</option>
			</select><br>
                        <input type="submit" value="calcular">
                </form>
        </body>
</html>

File: sprint3python/calculadora.py
from operaciones import suma, resta, multiplicacion, division

def calculadora():
    while True:
        try:
            a = float(input("Introduce el primer número: "))
            b = float(input("Introduce el segundo número: "))

            print("\nElige una operación: ")
            print("1. Suma")
            print("2. Resta")
            print("3. Multiplicación")
            print("4. División")
            opcion = input("Introduce el número de la operación (1, 2, 3, 4): ")

            if opcion == "1":
                resultado = suma(a, b)
                print(f"Resultado de la suma: {resultado}")
            elif opcion == "2":
                resultado = resta(a, b)
                print(f"Resultado de la resta: {resultado}")
            elif opcion == "3":
                resultado = multiplicacion(a, b)
                print(f"Resultado de la multiplicación: {resultado}")
            elif opcion == "4":
                resultado = division(a, b)
                print(f"Resultado de la división: {resultado}")
            else:
                print("Opción no válida.")
                continue

            pregunta = input("Otra operación? (s/n)")
            if pregunta != "s":
                print("Cálculo finalizado")
                break

        except ValueError:
            print("Error: Por favor, introduce un número válido.")
        except Exception as e:
            print(f"Un error ha aparecido: {e}")

if __name__ == "__main__":
    calculadora()